engine/src/main/java/org/pentaho/di/shared/PassthroughSharedObjectsManager.java

```diff
@@ -0,0 +1,90 @@
+package org.pentaho.di.shared;
+
+import org.pentaho.di.cluster.SlaveServer;
+import org.pentaho.di.core.database.DatabaseMeta;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.exception.KettleXMLException;
+import org.pentaho.di.shared.SharedObjectInterface;
+import org.pentaho.di.shared.SharedObjectsIO;
+import org.pentaho.di.shared.SharedObjectsManagementInterface;
+import org.pentaho.di.shared.SharedObjectsManager;
+import org.w3c.dom.Node;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This Manager that does not cache anything. Complete passthrough to the provided SharedObjectsIO instance.
+ *
+*/
+public class PassthroughSharedObjectsManager implements SharedObjectsManagementInterface {
+
+  private final SharedObjectsIO sharedObjectsIO;
+  public PassthroughSharedObjectsManager( SharedObjectsIO sharedObjectsIO ) {
+    this.sharedObjectsIO = sharedObjectsIO;
+  }
+
+  @Override
+  public void add( String type, SharedObjectInterface sharedObjectInterface ) throws KettleException {
+    Node node = SharedObjectsManager.toNode( type, sharedObjectInterface );
+    sharedObjectsIO.saveSharedObject(  type, sharedObjectInterface.getName(), node );
+  }
+
+  @Override
+  public SharedObjectInterface getSharedObject( String type, String name) throws KettleException {
+    Node node = sharedObjectsIO.getSharedObject( type, name );
+    if ( node == null ) {
+      return null;
+    }
+
+    return getSharedObject( type, node );
+  }
+
+  @Override
+  public List<SharedObjectInterface> getSharedObjects( String type ) throws KettleException {
+    Map<String, Node> nodeMap = sharedObjectsIO.getSharedObjects( type );
+    List<SharedObjectInterface> result = new ArrayList<>( nodeMap.size() );
+
+    for ( Node node : nodeMap.values() ) {
+      result.add( getSharedObject( type, node ) );
+    }
+    return result;
+  }
+
+  @Override
+  public void clear( String type ) throws KettleException {
+    sharedObjectsIO.clear( type );
+  }
+
+  @Override
+  public void removeSharedObject( String type, SharedObjectInterface sharedObjectInterface ) throws KettleException {
+    removeSharedObject( type, sharedObjectInterface.getName() );
+  }
+  @Override
+  public void removeSharedObject(String type, String sharedObjectName) throws KettleException {
+    sharedObjectsIO.delete( type, sharedObjectName );
+  }
+
+  private SharedObjectInterface getSharedObject(String type, Node node ) throws KettleException {
+    SharedObjectInterface sharedObjectInterface = null;
+    switch ( SharedObjectsIO.SharedObjectType.valueOf( type.toUpperCase() ) ) {
+      case CONNECTION:
+        sharedObjectInterface = new DatabaseMeta( node );
+        break;
+      case SLAVESERVER:
+        sharedObjectInterface = new SlaveServer( node );
+        break;
+      case PARTITIONSCHEMA:
+        // partitionSchemas
+        break;
+      case CLUSTERSCHEMA:
+        // clusterSchemas
+        break;
```

Comment:
I don't love this style. It would be better if we didn't have hard-coded logic for each type.